terraform/modules/task-definitions/publishing-api/main.tf

```
@@ -59,115 +55,130 @@ data "aws_secretsmanager_secret" "sentry_dsn" {
-        { "name" : "PORT", "value" : "80" },
-        { "name" : "RABBITMQ_HOSTS", "value" : "rabbitmq.${var.govuk_app_domain_internal}" },
-        { "name" : "RABBITMQ_USER", "value" : "publishing_api" },
-        { "name" : "RABBITMQ_VHOST", "value" : "/" },
-        { "name" : "REDIS_HOST", "value" : var.redis_host },
-        { "name" : "REDIS_PORT", "value" : tostring(var.redis_port) },
-        { "name" : "REDIS_URL", "value" : "redis://${var.redis_host}:${var.redis_port}" },
-        { "name" : "RAILS_ENV", "value" : "production" },
-        { "name" : "STATSD_PROTOCOL", "value" : "tcp" },
-        { "name" : "STATSD_HOST", "value" : var.statsd_host },
-        { "name" : "UNICORN_WORKER_PROCESSES", "value" : "8" }
-      ],
-      "dependsOn" : [{
-        "containerName" : "envoy",
-        "condition" : "START"
-      }],
-      "logConfiguration" : {
-        "logDriver" : "awslogs",
-        "options" : {
-          "awslogs-create-group" : "true",
-          "awslogs-group" : "awslogs-fargate",
-          "awslogs-region" : "eu-west-1",
-          "awslogs-stream-prefix" : "awslogs-${local.service_name}"
-        }
-      },
-      "mountPoints" : [],
-      "portMappings" : [
-        {
-          "containerPort" : 80,
-          "hostPort" : 80,
-          "protocol" : "tcp"
-        }
-      ],
-      "secrets" : [
-        {
-          "name" : "CONTENT_STORE_BEARER_TOKEN",
-          "valueFrom" : data.aws_secretsmanager_secret.content_store_bearer_token.arn
-        },
-        {
-          "name" : "DATABASE_URL",
-          "valueFrom" : data.aws_secretsmanager_secret.database_url.arn
-        },
-        {
-          "name" : "DRAFT_CONTENT_STORE_BEARER_TOKEN",
-          "valueFrom" : data.aws_secretsmanager_secret.draft_content_store_bearer_token.arn
-        },
-        {
-          "name" : "EVENT_LOG_AWS_SECRET_KEY",
-          "valueFrom" : data.aws_secretsmanager_secret.event_log_aws_secret_key.arn
-        },
-        {
-          "name" : "GDS_SSO_OAUTH_ID",
-          "valueFrom" : data.aws_secretsmanager_secret.oauth_id.arn
-        },
-        {
-          "name" : "GDS_SSO_OAUTH_SECRET",
-          "valueFrom" : data.aws_secretsmanager_secret.oauth_secret.arn
-        },
-        {
-          "name" : "RABBITMQ_PASSWORD",
-          "valueFrom" : data.aws_secretsmanager_secret.rabbitmq_password.arn
-        },
-        {
-          "name" : "ROUTER_API_BEARER_TOKEN",
-          "valueFrom" : data.aws_secretsmanager_secret.router_api_bearer_token.arn
-        },
-        {
-          "name" : "SECRET_KEY_BASE",
-          "valueFrom" : data.aws_secretsmanager_secret.secret_key_base.arn
-        },
-        {
-          "name" : "SENTRY_DSN",
-          "valueFrom" : data.aws_secretsmanager_secret.sentry_dsn.arn
-        }
-      ]
-    }
-  ]
+  container_definitions = flatten(
+    concat(
+      [module.web.definition],
+      [for worker in module.worker : worker.definition]
```

I think this simplifies to

```suggestion
      module.worker[*].definition
```

---

Thanks, I wasn't aware of that syntax!

---

Terraform somewhat confusingly calls it "[splat](https://www.terraform.io/docs/configuration/expressions.html#splat-expressions)", which is a bit confusing because it's a different thing than the "splat" in Ruby or Powershell 😐 I suspect it was inspired by `foo[].bar` in jq.